### PR TITLE
Add scenario test 1.5.2.10 and 1.5.2.11

### PR DESCRIPTION
**Pull request by ruwiniwj**
## Purpose
Add scenario test 1.5.2.10 and 1.5.2.11

### CODE REVIEW

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java [hunk 1]
```diff
@@ -293,11 +280,10 @@ public void destroy() throws Exception {
                 PUBLISHER_CREATOR_USERNAME);
         apiPublisher.deleteAPI("APIVisibility_tagsDistinctRoles2", apiVersion,
                 PUBLISHER_CREATOR_USERNAME);
-//        todo uncomment when 1.5.2.10 is fixed
-//        apiPublisher.deleteAPI("APIVisibility_apiByTag1", apiVersion,
-//                PUBLISHER_CREATOR_USERNAME);
-//        apiPublisher.deleteAPI("APIVisibility_apiByTag2", apiVersion,
-//                PUBLISHER_CREATOR_USERNAME);
+        apiPublisher.deleteAPI("APIVisibility_apiByTag1", apiVersion,
+                PUBLISHER_CREATOR_USERNAME);
```
Comment (vithu30) [inline product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java]: Can't we add this in single line.

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java [hunk 2]
```diff
@@ -210,38 +214,21 @@ public void testVisibilityOfTagsUsedByMultipleAPIsWithDistinctRoles() throws Exc
...
-//        verifyResponse(apisWithTagResponse);
-//        assertTrue(apisWithTagResponse.getData().contains(apiName), "API \'" + apiName + "\' is not visible " +
-//                "to the user with restricted role");
-//        assertTrue(!apisWithTagResponse.getData().contains("APIVisibility_apiByTag1"),
-//                "API \'APIVisibility_tagsDistinctRoles1\' is visible to the user without the restricted role");
-//    }
+    @Test(description = "1.5.2.10", dependsOnMethods = {"testVisibilityOfTagsUsedByMultipleAPIsWithDistinctRoles"})
+    public void testListingOfRestrictedAPIsByTags() throws Exception {
+        String tag = "tagsDistinctRoles";
+
```
Comment (chamindias) [inline product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java]: Additional newline?

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java (hunk 1 above)
Comment (ruwiniwj) [reply product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java]: Fixed with https://github.com/wso2/product-apim/pull/5248

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java (hunk 2 above)
Comment (ruwiniwj) [reply product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java]: Fixed with #5248

**Pull request merged**